services/webnn/ort/graph_builder_ort.cc

```diff
@@ -241,15 +241,17 @@ std::string GraphBuilderOrt::GetOperandName(uint64_t operand_id) {
   }
 }
 
-std::string GraphBuilderOrt::CreateInitializerAsRawData(
-    base::span<const uint32_t> shape,
-    base::span<const uint8_t> data,
-    OperandDataType data_type) {
+template <typename T>
+std::string GraphBuilderOrt::CreateInitializer(base::span<const uint32_t> shape,
+                                               base::span<const T> data,
```

Comment:
It can still take `base::span<const uint8_t>`, caller can simply use `base::as_byte_span` to convert different data types.

Reply:
I find reference codes https://source.chromium.org/chromium/chromium/src/+/main:services/webnn/tflite/graph_builder_tflite.cc;l=1059;bpv=1;bpt=1?q=GraphBuilderTflite::SerializeTensorWithBuffer&ss=chromium%2Fchromium%2Fsrc:services%2Fwebnn%2F
, it uses template just like what I use here. Not sure which one is better?

Reply:
The reference code still uses `as_byte_span()`. It uses template type to encapsulate the handling of floating-point type that would simplify user code. I am fine with that usage.

Reply:
Another open: can we remove the `OperandDataType` parameter? We can map it like https://source.chromium.org/chromium/chromium/src/+/main:services/webnn/tflite/graph_builder_tflite.cc;l=1076;bpv=0;bpt=1

Reply:
> Another open: can we remove the `OperandDataType` parameter? We can map it like https://source.chromium.org/chromium/chromium/src/+/main:services/webnn/tflite/graph_builder_tflite.cc;l=1076;bpv=0;bpt=1

Sounds good, do it in a separate PR?

Reply:
> do it in a separate PR

Sounds good!